src/geometry/GeoInterface.hh

```diff
@@ -32,14 +38,31 @@ namespace celeritas
  * __NVCC__ compiler) we can't simply declare this pointer to be in the \c cuda
  * or \c cxx explicit namespaces.
  */
-struct GeoParamsPointers
+template<Ownership W, MemSpace M>
+struct GeoParamsData
 {
-    const vecgeom::VPlacedVolume* world_volume = nullptr;
+    using PlacedVolumeT = typename detail::VGTraits<M>::PlacedVolume;
+
+    const PlacedVolumeT* world_volume = nullptr;
+    int                  max_depth    = 0;
```

Comment:
Those are "tied together", i.e. you probably don't want to modify one without the other.  Are the GeomParamsData "reused" or should those 2 members be `const`?

Reply:
Everywhere, `Params` data are always set exactly once during the constructor, then the "reference" (which actually copy these non-Collection values) are copied on host, and the data's copied to device. We can't truly make the data const without adding extra copy constructors and going through a whole lot of other rigamarole -- but in practice (since the CollectionMirror is assigned only once) they're effectively const.